rules/prefer-string-slice.js

```diff
@@ -10,6 +10,24 @@ const argumentsVariable = templates.spreadVariable();
 const substrCallTemplate = templates.template`${objectVariable}.substr(${argumentsVariable})`;
 const substringCallTemplate = templates.template`${objectVariable}.substring(${argumentsVariable})`;
 
+const getNumericValue = node => {
+	if (node.type === 'Literal' && typeof node.value === 'number') {
+		return node.value;
+	}
+
+	if (node.type === 'UnaryExpression' && node.operator === '-') {
+		return 0 - getNumericValue(node.argument);
```

Comment:
this  0 should  remove

Reply:
also, do we really want recury this?  just number and negative number should enough

Reply:
> this 0 should remove

It's usually useful to explain *why* if it's not immediately obvious.

Reply:
Why should the `0` be removed?

Reply:
sorry, I was not explaining clear, this is negative number, right? 

`0 - 1` is the same as `-1`

am I missing anything?

Reply:
It's a positive number `Literal` wrapped in a `UnaryExpression` with a `-` operator, so the `0 -` part is to convert the positive number into its correct negative value. I do think I have added tests for that?

Reply:
I know that `-1` is parsed as `UnaryExpression`and `operator:-` and `argument:1`

so `return 0 - node.argument` equals to `return - node.argument`.

I am get confusing, am I wrong?